### DISCUSSION TITLE
Android FHIR peer-to-peer data transfer: Nearby Connections API issues

### DISCUSSION THREAD

**Discussion by ekigamba** [google/android-fhir#638] [Ideas]
Nearby Connections API - https://developers.google.com/nearby/connections/android

## Nearby Connections API shortcomings

On OpenSRP, we have been using the Nearby Connections API and faced some technical issues while in the field:

- Taking a long time to search devices
- Connections sometime dropping when transferring for more than a few seconds (especially for devices that had lots of images)

Can anyone shed light on why this is happening? Is this acceptable/expected?

cc @rkodev @jingtang10

**Comment by rkodev** [comment]
@ekigamba 
The device behavior from a qualitative rather than quantitative perspective are 

| Device  | Behaviour |
| ------------- | ------------- |
| [Blu Grand Energy](https://www.gsmarena.com/blu_grand_energy-8556.php) | Connections were somewhat stable but still disconnected |
| [Blu J1](https://www.gsmarena.com/blu_studio_j1-8735.php) |  Device frequently disconnects during P2P |
| [Samsung A 10](https://www.gsmarena.com/samsung_galaxy_a10-9580.php)  | Gave a relatively stable connection. Did not disconnect as Frequently  |

**Reply by ekigamba** [reply]
Thanks @rkodev

**Reply by ekigamba** [reply]
Feedback from the Reveal team
- Sometimes it takes long to find devices
